src/geowombat/ml/classifiers.py

```diff
@@ -85,14 +85,14 @@ def _prepare_labels(self, data, labels, col, targ_name):
     def _stack_it(data):
         return data.stack(sample=("x", "y", "time")).T
 
-    # @staticmethod
     def _prepare_predictors(self, data, targ_name):
 
         X = self._stack_it(data)
 
         # drop nans
         try:
             Xna = X[~X[targ_name].isnull()]
+            Xna = X[X[targ_name] != X.attrs["nodatavals"][0]]
```

Comment:
This was the issue, 0s were not being treated as missing values 
Not sure if this is the best way to pull the nodata values. Is the attribute updated with gw.open?

Reply:
X is an unstacked array of all values:

|band1|band2|band3|
|-------|-----|---|
| 0|0|0|
| 0|0|0|
| 1234|8372|2357|
| 0847|8475|3243|

Reply:
You should now be able to do:
```python
Xna = X.gw.mask_nodata()
```

Reply:
@jgrss  Not sure I can use that here. we are removing missing values from the X[targ_name] dimension which is the rasterized training data.  I could restructure things so that I mask nodata, and move over the nans to the rasterized training data, then remove them, but not sure it makes it more elegant.  If you have other thoughts let me know. 

In particular  I feel like this isn't the best solutions for getting the no data value `X.attrs["nodatavals"][0]]` is there a better way to call that?

Reply:
How about we add a property called `nodata`? 

```python
data.gw.nodata
```

Reply:
See #208